streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -785,12 +789,13 @@ private UserTaskAssignmentListener assignTasksToClients(final Cluster fullMetada
             );
             final org.apache.kafka.streams.processor.assignment.TaskAssignor assignor = userTaskAssignor.get();
             final TaskAssignment taskAssignment = assignor.assign(applicationState);
-            processStreamsPartitionAssignment(clientMetadataMap, taskAssignment);
             final AssignmentError assignmentError = validateTaskAssignment(applicationState, taskAssignment);
-            userTaskAssignmentListener = (GroupAssignment assignment, GroupSubscription subscription) -> {
+            processStreamsPartitionAssignment(clientMetadataMap, taskAssignment);
+            userTaskAssignmentListener = (assignment, subscription) -> {
                 assignor.onAssignmentComputed(assignment, subscription, assignmentError);
             };
```

Comment:
Really doesn't matter, this is just an fyi, but you don't need the brackets for one-line lambdas and we typically use the inline/bracket-less style when possible in Streams code

```suggestion
            userTaskAssignmentListener = (assignment, subscription) -> assignor.onAssignmentComputed(assignment, subscription, assignmentError)
```